server_list/models.py

```diff
@@ -0,0 +1,266 @@
+
+	# Server sent connection address
+	address = db.Column(db.String, nullable=False)
+	port = db.Column(db.Integer, nullable=False, default=30000)
+
+	# IP address announcement was received from
+	announce_ip = db.Column(db.String, nullable=False)
+
+	# Name of server software.  E.g. "minetest"
+	server_id = db.Column(db.String, nullable=True)
+
+	# List of player names, one per line
+	clients = db.Column(db.String, nullable=True)
+
+	# Highest number of clients ever seen
+	clients_top = db.Column(db.Integer, nullable=False)
+
+	# Maximum number of allowed clients
+	clients_max = db.Column(db.Integer, nullable=False)
+
+	# First time that we received an announcement from this server
+	first_seen = db.Column(db.DateTime, default=datetime.utcnow, nullable=False)
+
+	# Time that server sent "start" announcement.
+	# This can be used to calculate the current uptime.
+	start_time = db.Column(db.DateTime, default=datetime.utcnow, nullable=False)
+
+	# Time of most recent update request
+	last_update = db.Column(db.DateTime, default=datetime.utcnow, nullable=False)
+
+	# Amount of time that we've seen the server up for, in seconds
+	total_uptime = db.Column(db.Float, nullable=False)
+
+	# Most recent time that the server went down
+	down_time = db.Column(db.DateTime, nullable=True)
+
+	# Server sent value for age of world.
+	# Should nearly match uptime on a server that always announces.
+	game_time = db.Column(db.Integer, nullable=False)
+
+	# Server sent value based on sever loop timing
+	lag = db.Column(db.Float, nullable=True)
+
+	# Ping time in seconds
+	ping = db.Column(db.Float, nullable=False)
+
+	# List of enabled mods, one per line
+	mods = db.Column(db.String, nullable=True)
+
+	# Server release version
+	version = db.Column(db.String, nullable=False)
+
+	# Supported protocol versions
+	proto_min = db.Column(db.Integer, nullable=False)
+	proto_max = db.Column(db.Integer, nullable=False)
+
+	game_id = db.Column(db.String, nullable=False)
+
+	# Mapgen name
+	mapgen = db.Column(db.String, nullable=True)
+
+	# Server landing page URL
+	url = db.Column(db.String, nullable=True)
+
+	# Privileges granted to new players by default
+	default_privs = db.Column(db.String, nullable=True)
+
+	name = db.Column(db.String, nullable=False)
+
+	description = db.Column(db.String, nullable=False)
+
+	# Roughly the average number of players on the server
+	popularity = db.Column(db.Float, nullable=False)
+
+	# Continent determined from IP
+	geo_continent = db.Column(db.String(2), nullable=True)
+
+	# Flags
+	creative = db.Column(db.Boolean, nullable=False)
+	is_dedicated = db.Column(db.Boolean, nullable=False)
+	damage_enabled = db.Column(db.Boolean, nullable=False)
+	pvp_enabled = db.Column(db.Boolean, nullable=False)
+	password_required = db.Column(db.Boolean, nullable=False)
+	rollback_enabled = db.Column(db.Boolean, nullable=False)
+	can_see_far_names = db.Column(db.Boolean, nullable=False)
```

Comment:
Better to keep them as bools, better third normal form then

Reply:
I considered this, but decided to leave them separate to avoid having extra (de)serialization logic.  This also lets you query by these fields easily.